### ISSUE TITLE
Block Element doesn't appear in the context menu of a tab under specific circumstances

### ISSUE THREAD

**Issue by ghost** [uBlockOrigin/uBlock-issues#151]
### Prerequisites

- [x] I verified that this is not a filter issue
    - Filter issues MUST be reported at [filter issue tracker](https://github.com/uBlockOrigin/uAssets/issues)
    
- [x] This is not a support issue or a question
    - Support issues and questions are handled at [/r/uBlockOrigin](https://old.reddit.com/r/uBlockOrigin/)
    
- [x] I performed a cursory search of the issue tracker to avoid opening a duplicate issue
    - Your issue may already be reported.
- I tried to reproduce the issue when...
    - [x] uBlock Origin is the only extension
    - [ ] uBlock Origin with default lists/settings
    - [ ] using a new, unmodified browser profile
- [x] I am running the latest version of uBlock Origin
- [x] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I report is not a normal behavior

### Description

Sometimes if you right click and accidently click open a new Window instead of a new tab for a link and if you happen to quickly close that new Window and thereafter right click to open the context menu on the already opened tab, Block Element will not appear, so this is more of a visual issue under the above mentioned circumstances. 

### A specific URL where the issue occurs

`https://www.google.com/`

### Steps to Reproduce

1. Close all opened tabs and windows and browse to google.com in a new tab.
2. Right click on **Gmail** and select **Open link in a new Window** which will spawn a new Window, so **quickly** close it(as fast as you can) before it loads.
3. Now back on the already opened tab of `google.com`, right click to open the context menu and notice Block Element is missing.

### Expected behavior:

Block Element would appear when context menu of the tab is opened.

### Actual behavior:

Block Element is missing from the context menu.

### Supporting Evidence

Chrome 70

https://i.gyazo.com/27e5e60c750d0794ab1ff232ec94d57a.mp4

Firefox 63

https://i.gyazo.com/b3cc16d203426f59e2e718a8e0837b2b.mp4

### Your environment

* uBlock Origin version: 1.16.15b6
* Browser Name and version: Chrome 70/Firefox 63
* Operating System and version: Win 10 x64 1807 build

### PS 

It was difficult to reproduce on Firefox, so it may take you some tries. I have attached videos as supporting evidence so as to provide some visual aide in reproduction.

**Comment by uBlock-user** [comment]
Also happens when you open a tab by middle click.

STR

- Browse to `google.com` and middle click on `Gmail`, as soon a new tab opens, click it on the tab and close the tab

- Back on the `google.com` tab right click and find Block Element missing from the context menu.

**Comment by u-RraaLL** [comment]
After opening a link in a new tab, be it via the context menu or via middle-clicking, uBO's context-menu item will never appear if the connection is interrupted. Even by uBO itself.

STR:

1. Add `||reddit.com^|$document` to "My filters".
2. Open uBO's sub.
3. Middle-Click the link to reddit's home.
4. Right-click anywhere on the page.
5. See no context menu item for uBO.
6. Repeat 4-5 as long as you want.
7. Switch the tab or refresh the page to bring back the context menu.

https://user-images.githubusercontent.com/92610420/179397010-90bc9991-f1a9-4fac-bbdb-d03edf143351.mp4

This is a new browser profile with uBO just installed and updated.

The missing context menu also occurs on lagging websites. I'm guessing before the initial response from the website reaches the browser. So immediately after opening a new tab on any website - normally not an issue since that'll usually last only under/about 1s, but if a website/server is lagging, it can take up to a minute. Or more, if the connection fails.
One such website is `https://rmz.cr`, but the lagging occurs at random times so it's not that easy to reproduce.

But whether the connection is lagging or blocked by uBO, the issue is probably the same.

~~
Originally posted as a comment to my (related?) issue #2085, but this one seems like a better place. That issue is tagged as a FF only bug, but this is reproducible on both FF and Chrome.

The only difference between them I've noticed is that if I use picker via the popup and close it, the context menu item will come back on Chrome, while it won't on FF.

The main problem with the disappearing item when the connection in the new tab is interrupted remains in both.

**Comment by gwarser** [comment]
@u-RraaLL please create new issue.

**Comment by uBlock-user** [comment]
@u-RraaLL cannot reproduce on Firefox 127, can you now ?

**Comment by garry-ut99** [comment]
> Context menu entry do not appear when iframe src = `about:blank`

That's already another separate issue, a duplicate of:
- https://github.com/uBlockOrigin/uBlock-issues/issues/283

**Comment by gorhill** [comment]
> top right rectangular box

I don't see such thing.

**Comment by gorhill** [comment]
![image](https://github.com/user-attachments/assets/a17216e0-b93d-4247-aef7-b3acf4cec220)

**Comment by gorhill** [comment]
> Hide placeholders of blocked elements setting was disabled

Still works fine in Chromium, but not in Firefox. I stepped through the code and uBO does tell Firefox to add a "Block element..." item to the menu, but the browser is not doing it, so this appears to be a Firefox-specific quirk.

**Comment by gorhill** [comment]
Not really, it's Firefox not enforcing uBO's context menu entries, I currently don't have the time to track down the issue in Firefox's code base for such a small issue.

**Comment by garry-ut99** [comment]
Are you sure that it's Firefox specifc, because the issue can be reproduced on Chromium too: in browser inspector pane, by changing manually the iframe's src attribute to:  `src="about:blank"` (or to `src=""`), then uBO menu doesn't appear anymore in that frame.

**Comment by gorhill** [comment]
The `src` was not `about:blank` in the reported case:

![image](https://github.com/user-attachments/assets/05fe8bc9-e78f-44b1-928d-44783997ab54)

For the `about:blank` case, using `'<all_urls>'` instead at [this line](https://github.com/gorhill/uBlock/blob/1.59.1b10/src/js/contextmenu.js#L163) seems to work. This still doesn't solve the [reported case above](https://github.com/uBlockOrigin/uBlock-issues/issues/151#issuecomment-2323178735) for Firefox, which is expected since the `src` is not `about:blank`.

**Comment by gorhill** [comment]
The screenshot I posted is the result of following your steps, using Firefox 129.0. I don't see `about:blank`, I see `https://discordapp.com/widget?id=556038524151791616&theme=dark`

**Comment by gorhill** [comment]
> you will see this

Very well. The proposed change to `'<all_urls>'` still doesn't make a difference in Firefox, so there is something specific to Firefox which prevent the context menu entry from appearing no matter what.

**Comment by garry-ut99** [comment]
Moreover, unlike in the previous case from my last comment, this time, on `https://fc-lc.xyz/ltio` , on uBO 1.46.0:
- (**A**[1]) _uBlock Origin_ ==> _Block Element_ : works since the beginning (Element Picker appears)
- (**A**[2])  _uBlock Origin_ ==> _Block Element in iFrame_ : Element Picker still doesn't appear

**Comment by gorhill** [comment]
Please provide browser name for when you state it works/doesn't works.

**Comment by garry-ut99** [comment]
(As for me) The same result for both Firefox/Chromium. (that's why I didn't mention browser name, but I understand you just want to make sure).

**Comment by gorhill** [comment]
> a case where I don't have to _disable Hide placeholders of blocked elements_

When I change [this line](https://github.com/gorhill/uBlock/blob/1.59.1b13/src/js/contextmenu.js#L163) to:

    documentUrlPatterns: [ '<all_urls>' ],

It solves the issue in Chromium. It does not solve the issue in Firefox. To solve the issue in Firefox, I need to completely remove the line, but then this brings back [this issue](https://github.com/uBlockOrigin/uBlock-issues/issues/1093#issuecomment-1423342620) -- which I argue is a Firefox issue, but nevertheless I do not want to bring it back.

The workaround is trivial : use the _Enter element picker mode_ from the popup panel or use assigned keyboard shortcut if any to enter element picker mode.

Since I can't fix the issue in Firefox, I consider it's not worth to make any change and advise to use the workaround.

**Comment by gorhill** [comment]
`*://*/*` does not match `about:blank`.

**Comment by garry-ut99** [comment]
Doesn't work either. Besides, AdGuard's and AdBlocker Ultimate's context menus works on `about:blank`, but suffer https://github.com/uBlockOrigin/uBlock-issues/issues/1093 as a side-effect.
